lib/puppet/transaction/event_manager.rb

```diff
@@ -144,18 +144,39 @@ def process_callback(resource, callback, events)
       return false
     end
 
+    message = "Triggered '#{callback}' from #{events.length} events"
+
     resource.send(callback)
 
-    if not resource.is_a?(Puppet::Type.type(:whit))
-      resource.notice "Triggered '#{callback}' from #{events.length} events"
-    end
+    resource.notice message if not resource.is_a?(Puppet::Type.type(:whit))
+    event = create_callback_status_event(resource, callback, message) if not resource.is_a?(Puppet::Type.type(:whit))
```

Comment:
Since both are conditional on `whit`s, can you roll that into one branch?

```
resource.send(callback)

if not resource.is_a?(Puppet::Type.type(:whit))
  message = "Triggered '#{callback}' from #{events.length} events"
  resource.notice message
  event = create_callback_status_event(resource, callback, message)
end